goalc/compiler/compilation/Type.cpp

```diff
@@ -85,6 +85,75 @@ RegVal* Compiler::compile_get_method_of_object(RegVal* object,
   return deref->to_reg(env);
 }
 
+Val* Compiler::compile_format_string(const goos::Object& form,
+                                     Env* env,
+                                     std::string& fmt_template,
+                                     std::vector<RegVal*> args,
+                                     const std::string& out_stream) {
+  // Add first two format args
+  args.insert(args.begin(), compile_string(fmt_template, env, DEBUG_SEGMENT)->to_reg(env));
```

Comment:
Instead of putting `DEBUG_SEGMENT` here, you could do `get_parent_env_of_type<FunctionEnv>(env)->segment` which gets the segment that the enclosing function is in. In your case this will always be the `DEBUG_SEGMENT` because you pass in the `method_env`, but it's nice to make this more general.

The convention for most `compile` functions that generate static data is that they either explicitly take a segment or default to the enclosing function's segment.